currency/pairs.go

```diff
@@ -211,12 +211,17 @@ func (p Pairs) Remove(pair Pair) (Pairs, error) {
 }
 
 // Add adds a specified pair to the list of pairs if it doesn't exist
-func (p Pairs) Add(pair Pair) Pairs {
-	if p.Contains(pair, true) {
-		return p
+func (p Pairs) Add(pairs ...Pair) Pairs {
+	p2 := make(Pairs, len(pairs))
```

Comment:
suggestion: Can consolidate your idea a little bit more. Let me know if this works.

```go
merge := append(slices.Clone(p), pairs...)
var filterInt int
for x := len(p); x < len(merge); x++ {
	if !merge[:len(p)+filterInt].Contains(merge[x], true) {
		merge[len(p)+filterInt] = merge[x]
		filterInt++
	}
}
return merge[:len(p)+filterInt]
```